flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -450,6 +450,63 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "1111111111111111111111111111111111111111111111111111111111111111")
   }
 
+  @Test
+  def testIsNumeric(): Unit = {
```

Comment:
Maybe add invalid argument cases to `ScalarFunctionsValidationTest` as well?

Reply:
Originally what I meant is, since this function only supports string/varchar, let's have a test that specifies `ISNUMERIC(1L)` throws `ValidationException`. 

Regarding the usage of this in general, I think this is useful to chained with many other operators with strict type constrains. such as `CASE WHEN ISNUMERIC(...) THEN ... ELSE ...`, where the `THEN` clause requires some strict numeric vaues. That's why I was wondering if we should do a better support beyond just STRING/VARCHAR type. 

I will comment on the JIRA actually. thanks for bring this up.